blockchain/fullblocktests/generate.go

```diff
@@ -1457,48 +1541,102 @@ func Generate(includeLargeReorg bool) (tests [][]TestInstance, err error) {
 
 	// Create block with an invalid merkle root.
 	//
-	//   ... -> b46(14)
-	//                 \-> b51(15)
-	g.SetTip("b46")
-	g.NextBlock("b51", outs[15], ticketOuts[15], func(b *wire.MsgBlock) {
+	//   ... -> brs3(14)
+	//                 \-> bmf5(15)
+	g.SetTip("brs3")
+	g.NextBlock("bmf5", outs[15], ticketOuts[15], func(b *wire.MsgBlock) {
+		// Set the merkle root to an invalid hash.
 		b.Header.MerkleRoot = chainhash.Hash{}
 	})
 	g.AssertTipBlockMerkleRoot(chainhash.Hash{})
 	rejected(blockchain.ErrBadMerkleRoot)
 
+	// Create block with an invalid stake root.
+	//
+	//   ... -> brs3(14)
+	//                 \-> bmf6(15)
+	g.SetTip("brs3")
+	g.NextBlock("bmf6", outs[15], ticketOuts[15], func(b *wire.MsgBlock) {
+		// Set the stake root to an invalid hash.
+		b.Header.StakeRoot = chainhash.Hash{}
+	})
+	g.AssertTipBlockStakeRoot(chainhash.Hash{})
+	rejected(blockchain.ErrBadMerkleRoot)
+
+	// Create block with an invalid block size.
+	//
+	//   ... -> brs3(14)
+	//                 \-> bmf7(15)
+	g.SetTip("brs3")
+	g.NextBlock("bmf7", outs[15], ticketOuts[15], func(b *wire.MsgBlock) {
+		b.Header.Size++
+	})
+	rejected(blockchain.ErrWrongBlockSize)
+
+	// Create block with an invalid subsidy for a coinbase input.
+	//
+	//   ... -> brs3(14)
+	//                 \-> bmf8(15)
+	g.SetTip("brs3")
+	g.NextBlock("bmf8", outs[15], ticketOuts[15], func(b *wire.MsgBlock) {
+		b.Transactions[0].TxIn[0].ValueIn++
+	})
+	rejected(blockchain.ErrBadCoinbaseAmountIn)
+
+	// Create block with an invalid subsidy for a stakebase input.
+	//
+	//   ... -> brs3(14)
+	//                 \-> bmf9(15)
+	g.SetTip("brs3")
+	g.NextBlock("bmf9", outs[15], ticketOuts[15], func(b *wire.MsgBlock) {
+		b.STransactions[0].TxIn[0].ValueIn++
+	})
+	rejected(blockchain.ErrBadStakebaseAmountIn)
+
+	// Create block with a revocations count mismatch.
```

Comment:
s/with a revocations count mismatch/with a header that commits to more revocations than the block actually contains./